#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.999003516253968 -0.040592853787913045 -0.018552485913242597 1.1166394473339505
  position 0.0742259028429686 2.0355955102147854 0.9385578508106048
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 2 2
}
Solid {
  translation -1 0.05 0
  children [
    DEF WALL Shape {
      appearance PBRAppearance {
        baseColor 0.0261692 0 0.998199
        metalness 0
      }
      geometry Box {
        size 0.01 0.1 2
      }
    }
  ]
  name "Wall 1"
  boundingObject USE WALL
}
Solid {
  translation 0 0.05 -1
  rotation 0 1 0 1.57
  children [
    USE WALL
  ]
  name "Wall 2"
  boundingObject USE WALL
}
Solid {
  translation 1 0.05 0
  children [
    USE WALL
  ]
  name "Wall 3"
  boundingObject USE WALL
}
Solid {
  translation 0 0.05 1
  rotation 0 1 0 1.57
  children [
    USE WALL
  ]
  name "Wall 4"
  boundingObject USE WALL
}
Robot {
  translation 0 0.0395245 0
  rotation 0.999999999885702 -1.5056838075509958e-05 -1.3738716540516065e-06 0.008004473635758744
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position -7.61599926296928
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "Wheel 1"
        }
      ]
      endPoint Solid {
        translation 0.05999999934251677 -3.6466261934935902e-06 0.05000144893932337
        rotation -0.5270048283933623 0.5270067642611634 0.6667306662161725 1.9655022252592966
        children [
          DEF WHEELS Shape {
            appearance PBRAppearance {
              baseColor 0.154971 1 0.0374304
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "Wheel 1"
        boundingObject USE WHEELS
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    DEF CHASSIS Shape {
      appearance PBRAppearance {
        baseColor 0.986282 0.0463111 0.121553
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -7.615999834885261
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "Wheel 2"
        }
      ]
      endPoint Solid {
        translation 0.06000000065788258 -3.646631992277998e-06 -0.04999855107528264
        rotation -0.5270049660160477 0.5270069018892914 0.6667304486488338 1.9655025264275814
        children [
          USE WHEELS
        ]
        boundingObject USE WHEELS
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -7.6159992638774705
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "Wheel 3"
        }
      ]
      endPoint Solid {
        translation -0.06000000065748304 -3.6461562994685095e-06 0.05000145044468602
        rotation -0.5270048286129817 0.5270067644807844 0.6667306658689826 1.9655022257399921
        children [
          USE WHEELS
        ]
        name "solid(1)"
        boundingObject USE WHEELS
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -7.615999835194164
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "Wheel 4"
        }
      ]
      endPoint Solid {
        translation -0.059999999342117236 -3.646162098397027e-06 -0.04999854956992242
        rotation -0.5270049660913727 0.5270069019646171 0.6667304485297544 1.9655025265924504
        children [
          USE WHEELS
        ]
        name "solid(2)"
        boundingObject USE WHEELS
        physics USE WHEEL_PH
      }
    }
    DistanceSensor {
      translation 0 0 0.1
      rotation 0 1 0 -1.57
      children [
        Solid {
          children [
            DEF SENSOR Shape {
              appearance PBRAppearance {
                baseColor 0.0684367 0 0.998215
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
          boundingObject USE SENSOR
        }
      ]
      name "sensor 1"
      physics Physics {
      }
      lookupTable [
        0 1023 0
        0.02 1023 0
        0.3 0 0
      ]
      type "sonar"
      numberOfRays 2
      aperture 0.75
    }
  ]
  name "4 Wheel Robot"
  boundingObject USE CHASSIS
  physics Physics {
  }
  controller "Sonar_Interfacing"
}
Solid {
  translation -6.729777439854859e-19 0.09685977938060325 0.48
  rotation 0.8551838093621402 0.07333816877047228 0.5131102856172854 4.985193087016527e-17
  children [
    DEF BOX Shape {
      appearance PBRAppearance {
        baseColor 0.99968 0.959503 0.041062
        metalness 0
      }
      geometry Box {
        size 0.2 0.2 0.2
      }
    }
  ]
  boundingObject USE BOX
  physics Physics {
  }
  linearVelocity -3.001457785739191e-17 8.135720861485385e-06 5.7265897153697736e-18
  angularVelocity 2.664535259100373e-16 -6.129842099378921e-17 1.332267629550188e-16
}
